packages/lesswrong/server/resolvers/petrovDay2024Resolvers.ts

```diff
@@ -0,0 +1,51 @@
+import { PetrovDayLaunchs } from "@/lib/collections/petrovDayLaunchs";
+import { addGraphQLMutation, addGraphQLQuery, addGraphQLResolvers, addGraphQLSchema } from "../vulcan-lib";
+import { petrovDayLaunchCode } from "@/components/seasonal/PetrovDayButton";
+import PetrovDayActions from "@/lib/collections/petrovDayActions/collection";
+import { petrovBeforeTime } from "@/components/Layout";
+import { DatabaseServerSetting } from "../databaseSettings";
+import sample from "lodash/sample";
+
+const petrovFalseAlarmMissileCount = new DatabaseServerSetting<number[]>('petrovFalseAlarmMissileCount', [])
+const petrovRealAttackMissileCount = new DatabaseServerSetting<number[]>('petrovRealAttackMissileCount', [])
+
+const getIncomingCount = (incoming: boolean) => {
+  const currentHour = new Date().getHours();
+  const seed = currentHour + (incoming ? 24 : 0); // Different seed for each hour and incoming state
+
+  const missileCountArray = incoming ? petrovRealAttackMissileCount.get() : petrovFalseAlarmMissileCount.get();
+
+  return missileCountArray[seed % missileCountArray.length];
+}
+
+const PetrovDay2024CheckNumberOfIncomingData = `type PetrovDay2024CheckNumberOfIncomingData {
+  count: Int
+}`
+
+addGraphQLSchema(PetrovDay2024CheckNumberOfIncomingData);
+
+const petrovDay2024Resolvers = {
+  Query: {
+    async PetrovDay2024CheckNumberOfIncoming(root: void, args: void, context: ResolverContext) {
+      const startTime = new Date(petrovBeforeTime.get())
```

Comment:
This probably needs to be gated to only users who are allowed to check number of incoming (i.e. users with the petrov role?)

Reply:
I think it is?

Reply:
oh, you do it after everything else.  Seems iffy from a code structure perspective.  (Also I would do an early return if there's no `currentUser` in the context.)